Classes/Command/ResetCommand.php

```diff
@@ -0,0 +1,106 @@
+<?php
+namespace IchHabRecht\Filefill\Command;
+
+use Doctrine\DBAL\Connection;
+use IchHabRecht\Filefill\Repository\FileRepository;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+class ResetCommand extends Command
+{
+    /**
+     * @var \TYPO3\CMS\Core\Database\Connection
+     */
+    protected $connection;
+
+    /**
+     * @var FileRepository
+     */
+    protected $fileRepository;
+
+    public function __construct(string $name = null, Connection $connection = null, FileRepository $fileRepository = null)
+    {
+        parent::__construct($name);
+
+        $this->connection = $connection ?: GeneralUtility::makeInstance(ConnectionPool::class)->getConnectionForTable('sys_file');
+        $this->fileRepository = $fileRepository ?: GeneralUtility::makeInstance(FileRepository::class);
+    }
+
+    public function configure()
+    {
+        $this->setDescription('Deletes files fetched by filefill')
+            ->addOption(
+                'storage',
+                's',
+                InputOption::VALUE_OPTIONAL,
+                'Delete files from a specific storage only'
+            );
+    }
+
+    /**
+     * Executes the command
+     *
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $storage = $input->getOption('storage');
+
+        $queryBuilder = $this->connection->createQueryBuilder();
+        $expressionBuilder = $queryBuilder->expr();
+        $queryBuilder->getRestrictions()->removeAll();
+        $queryBuilder->count('*')
+            ->addSelect('f.storage', 's.name')
+            ->from('sys_file', 'f')
+            ->leftJoin(
+                'f',
+                'sys_file_storage',
+                's',
+                $expressionBuilder->eq('s.uid', $queryBuilder->quoteIdentifier('f.storage'))
+            )
+            ->where(
+                $expressionBuilder->eq(
+                    'f.missing',
+                    $queryBuilder->createNamedParameter(1, \PDO::PARAM_INT)
+                )
+
+            )
+            ->groupBy('f.storage')
+            ->orderBy('f.storage');
+
+        if ($storage !== null) {
+            $queryBuilder->andWhere(
+                $expressionBuilder->eq(
+                    'f.storage',
+                    $queryBuilder->createNamedParameter($storage, \PDO::PARAM_INT)
+                )
+            );
+        }
+
+        $statement = $queryBuilder->execute();
+
+        while ($row = $statement->fetch(\PDO::FETCH_NUM)) {
```

Comment:
if I understand this correctly, this resets missing files on all (or selected) storages, whether they are file fill enabled or not. is this correct?

Reply:
Yes. It is necessary to be able to allow fetching files by filefill.